src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/OptionsTests.cs

```diff
@@ -424,6 +432,43 @@ public static void Options_GetConverterForObjectJsonElement_GivesCorrectConverte
             GenericObjectOrJsonElementConverterTestHelper<JsonElement>("JsonElementConverter", element, "[3]");
         }
 
+        [Fact]
+        public static void Options_JsonSerializerContext_DoesNotFallbackToReflection()
+        {
+            var options = JsonContext.Default.Options;
+            JsonSerializer.Serialize(new WeatherForecastWithPOCOs(), options); // type supported by context should succeed serialization
+
+            var unsupportedValue = new MyClass();
+            Assert.Throws<InvalidOperationException>(() => JsonSerializer.Serialize(unsupportedValue, unsupportedValue.GetType(), JsonContext.Default));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize(unsupportedValue, options));
+        }
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public static void Options_JsonSerializerContext_DoesNotFallbackToReflection_GetConverter()
+        {
+            RemoteExecutor.Invoke(static () =>
+            {
+                var options = JsonContext.Default.Options;
+                JsonSerializer.Serialize(new WeatherForecastWithPOCOs(), options); // type supported by context should succeed serialization
+
+                var unsupportedValue = new MyClass();
+                Assert.Throws<NotSupportedException>(() => options.GetConverter(unsupportedValue.GetType()));
```

Comment:
this doesn't match previous behavior, correct? I'm ok with that but we should at least add test that if converter for unsupported value got added to options that also returns value

Reply:
Yes, you're right. I'm reverting this behavior and updating the test.